.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and Deploy to Github Pages
+
+on:
+  push:
+    branches:
+      - main # Here source code branch is `master`, it could be other branch
+
+jobs:
+  deploy:
+    runs-on: ubuntu-20.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: "0.4.10"
+          # mdbook-version: 'latest'
+
+      - uses: actions-rs/install@v0.1
+        with:
+          crate: mdbook-toc
+          version: latest
+
+#      - uses: actions-rs/install@v0.1
+#        with:
+#          crate: mdbook-mermaid
```

Comment:
Any specific reason why mermaid was disabled? Because we don't have any mermaid diagrams in our official docs? I don't for sure. So, just checking to be sure.

Reply:
No actually there was no specific reason, I can enable it.